### ISSUE TITLE
Quad9 - DoT and DoH not working

### ISSUE THREAD

**Issue by nickydd9** [TechnitiumSoftware/DnsServer#711]
I've noticed in the past couple days DoT and DoH forwarding to Quad9 does not seem to be working. When I go to "DNS Client" and test using any of the DoT or DoH servers for Quad9 it spins and then says: "Error! DnsClient failed to resolve the request 'cnn.com. A IN': request timed out."

If I use regular UDP to 9.9.9.9:53 it works just fine.

If I use Cloudflare 1.1.1.1:53, or Cloudflare DoT or Cloudflare DoH it also works just fine.

**Comment by ShreyasZare** [comment]
Thanks for the feedback. It seems to be a problem with your network which is causing this issue since both DoT and DoH are working here for all public DNS providers that are listed. It could be that your ISP is blocking well known DoT and DoH servers.

Check the DNS logs from the admin panel and see if there are any errors related to it.

**Comment by nickydd9** [comment]
I don't think my ISP (Rogers) is blocking based on this test below, but I will double check with them to be sure. 

![DoT](https://github.com/TechnitiumSoftware/DnsServer/assets/33927790/703e7efd-6a7c-4472-88b4-4b9f90bc0106)

Attached also is excerpt of log from admin panel.
[DNSLog.txt](https://github.com/TechnitiumSoftware/DnsServer/files/12489096/DNSLog.txt)

**Comment by ShreyasZare** [comment]
TCP tests alone wont tell if the service is being blocked. Try running `openssl s_client -connect 9.9.9.9:853` command and share the output here.

The DoT implementation does work which you can test with the [dnsclient.net](https://dnsclient.net/#Quad9%20Secure%20TLS%20%7Bdns.quad9.net%20(9.9.9.9%3A853)%7D/cnn.com/HTTPS/TLS/false) website that uses the same code.

**Comment by nickydd9** [comment]
Ok, thanks then. Here is the only output when I run that command:

![openssl](https://github.com/TechnitiumSoftware/DnsServer/assets/33927790/9de79db1-b28a-481f-b425-b60827cbe2be)

When I run the command using 1.1.1.1:853 instead, I get a much more complete output showing SSL session.

**Comment by ShreyasZare** [comment]
> Ok, thanks then. Here is the only output when I run that command:
> 
> ![openssl](https://user-images.githubusercontent.com/33927790/265013974-9de79db1-b28a-481f-b425-b60827cbe2be.png)
> 
> When I run the command using 1.1.1.1:853 instead, I get a much more complete output showing SSL session.

It seems to be case of blocking. The ISP is accepting the TCP connection but wont start the SSL handshake effectively causing a timeout error blocking the access. It could also be some other unknown network issue having that effect.

**Comment by nickydd9** [comment]
Ok, well I am on the phone with my ISP again. It's been a struggle to escalate this to a more senior technical team on their side. I have also opened a support ticket with Quad9 in case my public IP is blacklisted somewhere on their side, although not sure why it would be. 

As a quick test I bypassed all my home network gear and patched a computer directly into my ISP modem and still I don't see that SSL handshake happening, so at least I have ruled out my own gear. 

I'll update this issue when I know more. I may end up just switching to Cloudflare DoT if I can't get it working.
